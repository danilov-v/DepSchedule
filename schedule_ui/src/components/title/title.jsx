import React from "react";
import PropTypes from "prop-types";
import DatePicker from "react-datepicker";
import ru from "date-fns/locale/ru";

import "react-datepicker/dist/react-datepicker.css";
import "./title.scss";

export function Title({
  startDate,
  endDate,
  onChangeStartDate,
  onChangeEndDate,
}) {
  return (
    <div className="title">
      График развёртывания СУ в периуд с &nbsp;
      <DatePicker
        selected={startDate}
        dateFormat="dd/MM/yyyy"
        onChange={date => onChangeStartDate(date)}
        locale={ru}
        className="date_picker"
      />
      &nbsp; по &nbsp;
      <DatePicker
        selected={endDate}
        dateFormat="dd/MM/yyyy"
        onChange={date => onChangeEndDate(date)}
        locale={ru}
        className="date_picker"
      />
    </div>
  );
}

Title.propTypes = {
  onChangeStartDate: PropTypes.func.isRequired,
  onChangeEndDate: PropTypes.func.isRequired,
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date),
};
