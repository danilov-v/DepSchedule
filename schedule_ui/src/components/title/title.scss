@import "variables.scss";

.title {
  text-align: center;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  color: $color-grey-dark-3;

  .react-datepicker-popper {
    z-index: 10;
  }
}

.date_picker {
  font-size: 1.1rem;
  color: $color-grey-dark-3;
  border: 1px solid $color-grey-dark;
  border-radius: 5px;
}
