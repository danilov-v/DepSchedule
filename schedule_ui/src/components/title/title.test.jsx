import React from "react";
import { shallow } from "enzyme";

import { Title } from "./title";

describe("title.jsx", () => {
  describe("<Title />", () => {
    it("should render and match snapshot", () => {
      const component = shallow(
        <Title onChangeStartDate={jest.fn} onChangeEndDate={jest.fn} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
