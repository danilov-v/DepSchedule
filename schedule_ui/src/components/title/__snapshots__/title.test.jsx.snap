// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title.jsx <Title /> should render and match snapshot 1`] = `
<div
  className="title"
>
  График развёртывания СУ в периуд с  
  <DatePicker
    allowSameDay={false}
    className="date_picker"
    dateFormat="dd/MM/yyyy"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    inlineFocusSelectedMonth={false}
    locale={
      Object {
        "formatDistance": [Function],
        "formatLong": Object {
          "date": [Function],
          "dateTime": [Function],
          "time": [Function],
        },
        "formatRelative": [Function],
        "localize": Object {
          "day": [Function],
          "dayPeriod": [Function],
          "era": [Function],
          "month": [Function],
          "ordinalNumber": [Function],
          "quarter": [Function],
        },
        "match": Object {
          "day": [Function],
          "dayPeriod": [Function],
          "era": [Function],
          "month": [Function],
          "ordinalNumber": [Function],
          "quarter": [Function],
        },
        "options": Object {
          "firstWeekContainsDate": 1,
          "weekStartsOn": 1,
        },
      }
    }
    monthsShown={1}
    nextMonthButtonLabel="Next month"
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    readOnly={false}
    renderDayContents={[Function]}
    shouldCloseOnSelect={true}
    showMonthYearPicker={false}
    showTimeInput={false}
    showTimeSelect={false}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    withPortal={false}
  />
    по  
  <DatePicker
    allowSameDay={false}
    className="date_picker"
    dateFormat="dd/MM/yyyy"
    dateFormatCalendar="LLLL yyyy"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    inlineFocusSelectedMonth={false}
    locale={
      Object {
        "formatDistance": [Function],
        "formatLong": Object {
          "date": [Function],
          "dateTime": [Function],
          "time": [Function],
        },
        "formatRelative": [Function],
        "localize": Object {
          "day": [Function],
          "dayPeriod": [Function],
          "era": [Function],
          "month": [Function],
          "ordinalNumber": [Function],
          "quarter": [Function],
        },
        "match": Object {
          "day": [Function],
          "dayPeriod": [Function],
          "era": [Function],
          "month": [Function],
          "ordinalNumber": [Function],
          "quarter": [Function],
        },
        "options": Object {
          "firstWeekContainsDate": 1,
          "weekStartsOn": 1,
        },
      }
    }
    monthsShown={1}
    nextMonthButtonLabel="Next month"
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onInputClick={[Function]}
    onInputError={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    previousMonthButtonLabel="Previous Month"
    readOnly={false}
    renderDayContents={[Function]}
    shouldCloseOnSelect={true}
    showMonthYearPicker={false}
    showTimeInput={false}
    showTimeSelect={false}
    strictParsing={false}
    timeCaption="Time"
    timeInputLabel="Time"
    timeIntervals={30}
    withPortal={false}
  />
</div>
`;
